OrbitClientServices/ProcessManager.cpp

```diff
@@ -202,6 +205,19 @@ ErrorMessageOr<std::string> ProcessManagerImpl::LoadProcessMemory(
   return std::move(*response.mutable_memory());
 }
 
+ErrorMessageOr<std::string> ProcessManagerImpl::LoadNullTerminatedString(
+      int32_t pid, uint64_t address, uint64_t max_size){
+  auto errorOrString = LoadProcessMemory(pid, address, max_size);
+  if(errorOrString.has_value()) {
+    // Make sure the string is null terminated.
+    std::string str = errorOrString.value();
+    str[str.size()-1] = '\0';
+    errorOrString = str.c_str();
```

Comment:
why do you need this to be null-terminated?

Reply:
Modified the logic, please see comment.